AmplifyPlugins/Auth/AWSCognitoAuthPlugin/Dependency/AuthenticationProviderAdapter+DeleteUser.swift

```diff
@@ -0,0 +1,44 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Amplify
+#if COCOAPODS
+import AWSMobileClient
+#else
+import AWSMobileClientXCF
+#endif
+
+extension AuthenticationProviderAdapter {
+
+    func deleteUser(request: AuthDeleteUserRequest, completionHandler: @escaping (Result<Void, AuthError>) -> Void) {
+        awsMobileClient.deleteUser(signOut: false) { [weak self] error in
+            guard let error = error else {
+                let signOutOptions = AuthSignOutRequest.Options(globalSignOut: true)
+                let signOutRequest = AuthSignOutRequest(options: signOutOptions)
+                self?.signOut(request: signOutRequest) { result in
+                    switch result {
+                    case .success:
+                        completionHandler(.success(()))
+                        return
+                    case .failure(let error):
+                        completionHandler(.failure(AuthErrorHelper.toAuthError(error)))
+                        return
+                    }
+                }
+                return
+            }
+            if case .notSignedIn = error as? AWSMobileClientError {
+                let authError = AuthError.signedOut(AuthPluginErrorConstants.deleteUserSignOutError.errorDescription,
+                                                    AuthPluginErrorConstants.deleteUserSignOutError.recoverySuggestion,
+                                                    error)
+                completionHandler(.failure(authError))
+                return
+            }
+            completionHandler(.failure(AuthErrorHelper.toAuthError(error)))
```

Comment:
Does this give the right error message if the user is signed in by federating to Cognito Identity Pool?

We should test:
1. Sign in using username/password SRP (should delete user )
2. Sign in using hostedUI (should delete user )
3. Sign in via federation to Identity Pool ( should return error )

Reply:
Tested with all 3.  1 and 2 delete the user and I updated to error message for 3 to state that there is not a user signed in to Cognito User Pool.